R/bayesiannetworkanalysis.R

```diff
@@ -0,0 +1,1297 @@
+      groupNames <- matrix(unlist(options[["groupNames"]]), ncol = 2L, byrow = TRUE)
+      nGroups <- nrow(groupNames)
+      
+      idx <- match(variablesForColor[, 1L], groupNames[, 1L])
+      
+      groups <- vector("list", nGroups)
+      names(groups) <- groupNames[, 1L]
+      for (i in seq_len(nGroups))
+        groups[[i]] <- which(idx == i)
+      
+      nonEmpty <- lengths(groups) > 0L
+      groups <- groups[nonEmpty]
+      
+      if (options[["manualColors"]])
+        nodeColor <- groupNames[nonEmpty, 2L]
+    }
+  }
+  
+  # defaults
+  shape <- "circle"
+  edgeColor <- NULL
+  
+  # TODO: footnote if legend off and nodenames used
+  if (options[["showVariableNames"]] == "In nodes") {
+    nodeNames <- NULL
+    
+    if (nGraphs == 1) {
+      labels <- .unv(colnames(allNetworks$Network$graph))
+    } else {
+      labels <- .unv(colnames(allNetworks$`1`$graph))
+    }
+
+  } else {
+    
+    if (nGraphs == 1) {
+      nodeNames <- .unv(colnames(allNetworks$Network$graph))
+    } else {
+      nodeNames <- .unv(colnames(allNetworks$`1`$graph))
+    }
+    labels <- seq_along(nodeNames)
+    
+  }
+  
+  labels <- decodeColNames(labels)
+  
+  
+  if (options[["abbreviateLabels"]])
+    labels <- base::abbreviate(labels, minlength = options[["abbreviateNoChars"]])
+  
+  # do we need to draw legends?
+  if (!is.null(groups) || !is.null(nodeNames)) {
+    if (options[["showLegend"]] ==  "All plots") {
+      
+      allLegends <- rep(TRUE, nGraphs)
+      
+    } else if (options[["showLegend"]] ==  "In plot number: ") {
+      
+      if (options[["legendNumber"]] > nGraphs) {
+        
+        allLegends[nGraphs] <- TRUE
+        
+      } else if (options[["legendNumber"]] < 1L) {
+        
+        allLegends[1L] <- TRUE
+        
+      } else {
+        
+        allLegends[options[["legendNumber"]]] <- TRUE
+        
+      }
+    }
+  }
+  
+  names(allLegends) <- names(allNetworks) # allows indexing by name
+  
+  basePlotSize <- 320
+  legendMultiplier <- options[["legendToPlotRatio"]] * basePlotSize
+  height <- setNames(rep(basePlotSize, nGraphs), names(allLegends))
+  width  <- basePlotSize + allLegends * legendMultiplier
+  for (v in names(allNetworks))
+    structurePlotContainer[[v]] <- createJaspPlot(title = v, width = width[v], height = height[v])
+  
+  jaspBase::.suppressGrDevice({
+    
+    for (v in names(allNetworks)) {
+      
+      networkToPlot <- allNetworks[[v]]
+      
+      legend <- allLegends[[v]]
+      structurePlotContainer[[v]]$plotObject <- .bayesianNetworkAnalysisOneStructurePlot(
+        network    = networkToPlot,
+        options    = options,
+        layout     = layout,
+        groups     = groups,
+        labels     = labels,
+        legend     = legend,
+        shape      = shape,
+        nodeColor  = nodeColor,
+        nodeNames  = nodeNames
+      )
+    }
+  })
+  
+}
+
+.bayesianNetworkAnalysisNetworkPlot <- function(plotContainer, network, options) {
```

Comment:
wasn't it possible to reuse the code in the frequentist network analysis?

Reply:
You are completely right! I do have to make some minor adjustments but I can add an argument` method = "Bayesian"` I assume?

Reply:
That would be perfect!